src/main/java/com/auth0/net/Telemetry.java

```diff
@@ -4,50 +4,81 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.codec.binary.Base64;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 @SuppressWarnings("WeakerAccess")
-class Telemetry {
+public class Telemetry {
     static final String HEADER_NAME = "Auth0-Client";
 
     private static final String NAME_KEY = "name";
     private static final String VERSION_KEY = "version";
+    private static final String CORE_KEY = "core";
+    private static final String ENV_KEY = "env";
+    private static final String JAVA_KEY = "java";
 
     private final String name;
     private final String version;
+    private final String core;
+    private final Map<String, String> env;
+    private final String value;
 
-    Telemetry(String name, String version) {
-        this.name = name;
-        this.version = version;
-    }
-
-    public String getName() {
-        return name;
+    public Telemetry(String name, String version) {
+        this(name, version, null);
     }
 
-    public String getVersion() {
-        return version;
-    }
+    public Telemetry(String name, String version, String core) {
+        this.name = name;
+        this.version = version;
+        this.core = core;
 
-    public String getValue() {
-        Map<String, String> values = new HashMap<>();
+        Map<String, Object> values = new HashMap<>();
         if (name != null) {
             values.put(NAME_KEY, name);
         }
         if (version != null) {
             values.put(VERSION_KEY, version);
         }
-        if (values.isEmpty()) {
-            return null;
+
+        HashMap<String, String> tmpEnv = new HashMap<>();
+        tmpEnv.put(JAVA_KEY, Runtime.class.getPackage().getSpecificationVersion());
+        if (core != null) {
+            tmpEnv.put(CORE_KEY, core);
         }
-        String urlSafe = null;
+        this.env = Collections.unmodifiableMap(tmpEnv);
+        values.put(ENV_KEY, env);
+
+        String tmpValue;
         try {
             String json = new ObjectMapper().writeValueAsString(values);
-            urlSafe = Base64.encodeBase64URLSafeString(json.getBytes());
+            tmpValue = Base64.encodeBase64URLSafeString(json.getBytes());
         } catch (JsonProcessingException e) {
+            tmpValue = null;
             e.printStackTrace();
         }
-        return urlSafe;
+        value = tmpValue;
+    }
+
+    public String getName() {
```

Comment:
When are these getters used? Just for testing?

Reply:
Yep. I want to expose something so libraries overriding this can check everything is being replaced fine 👍